### PR TITLE
feat: Jenkinsfile + test playbook & roles

### PR DESCRIPTION
**Pull request by psturc**
### Motivation
https://issues.jboss.org/browse/AEROGEAR-2392

### Changes
* \+ Jenkinsfile, now the test is running on separate ocp slave
* added few variables instead of hardcoded names of objects
* added test playbook and roles
* fixed deprovision role

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                    """
+                }
+            } catch (Exception e) {
+                error "Build timed out"
+            }
+
+        }
+
+        stage('Add admin policy to serviceaccount "default"') {
+            sh script: "oc policy add-role-to-user admin --serviceaccount=default"
```
Comment (AdamSaleh) [inline Jenkinsfile:43]: Can't we have this on the slave already preset? This seems to be cluster-wide and not just per/project.

**Review by AdamSaleh — APPROVED** [review]
Thanks for clearing up the "oc policy add-role", I didn't know you don't need to specify the project-name in certain cases :-) 

I can't really comment on the APB changes.

File: Jenkinsfile (hunk 1 above)
Comment (matzew) [reply Jenkinsfile:43]: any comments, @psturc

File: roles/test-provision-unifiedpush-apb/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+- name: Check that the unifiedpush login webpage is accessible
+  uri:
+    url: "https://{{ ups_route.route.spec.host }}"
```
Comment (matzew) [inline roles/test-provision-unifiedpush-apb/tasks/main.yml:3]: should we perhaps also test that a PushAPP was created ?

File: Jenkinsfile (hunk 1 above)
Comment (psturc) [reply Jenkinsfile:43]: So this command adds the admin role to the `default` service account only for the scope of the project. It is not cluster-wide.

File: roles/test-provision-unifiedpush-apb/tasks/main.yml (hunk 2 above)
Comment (psturc) [reply roles/test-provision-unifiedpush-apb/tasks/main.yml:3]: test added

**Review by matzew — CHANGES_REQUESTED** [review]
getting:

```
{"error":"server_error","error_description":"The authorization server encountered an unexpected condition that prevented it from fulfilling the request.","state":"4f233504b27cce1278d0cb149c15a6e5:/"}
```

in the browser after trying to login.


I think this might be related to the "-route" name in the actual URL - which is generally odd

**Comment by pb82** [comment]
@psturc The Jenkins output looks a bit short:

```
[talog_unifiedpush-apb_PR-25-FOKLVIXRMFLNOQ2FIRMV3IDPGXSXV6F7DPEYDZ7V47JYXWEZWBGQ] Running shell script

+ oc delete project test-unifiedpush-apb-6-1522846272402

project "test-unifiedpush-apb-6-1522846272402" deleted
```

is that ok or does it not yet run any tests?

**Comment by psturc** [comment]
@pb82 The thing you're seeing is actually the last stage of the pipeline. You can see full log if you switch between stages in the "blue ocean" mode or switch go to classic Jenkins UI:  https://jenkins-wendy.ci.feedhenry.org/job/AeroGearCatalog/job/unifiedpush-apb/job/PR-25/6/console

@matzew that's strange, the test in Jenkins successfully passed. I'll try that locally.

**Comment by matzew** [comment]
@psturc why do we need that odd `-route` in the URL anyways ? :-)

**Comment by psturc** [comment]
@matzew ok ok, it doesn't make much sense to have that in the URL :^) Let's replace it

**Comment by psturc** [comment]
@matzew @pb82 all fixed now, test passed, and finally the login works.
apb is provisioned here https://psturc-apb-testing.skunkhenry.com:8443/console/project/test-ups/overview

**Comment by matzew** [comment]
merged: https://github.com/aerogearcatalog/unifiedpush-apb/commit/4615db0d51b1dc11323dbff93b26113f8d3a550c

**Pull request closed without merging**